### ISSUE TITLE
Loading OpenJFX broken after #5842

### ISSUE THREAD

**Issue by matthiasblaesing** [apache/netbeans#6004]
Testing a recent master build I noticed, that OpenJFX can't be loaded anymore. The problem can be reproduced on current master by running:

- `ant`
- `ant tryme`

![image](https://github.com/apache/netbeans/assets/2179736/99cf41bc-0c74-40bd-858e-16eb9432c0a2)

I bisected this from NB 17 to master and the commit, that introduced the problem is: 15811ec850cc3cfe50eeaffd96c2eae1065ccb9c, which is part of https://github.com/apache/netbeans/pull/5842.

Temporary fixes:

- revert 15811ec850cc3cfe50eeaffd96c2eae1065ccb9c
- comment out lines 1861, 1866, 1962, 1967 and 2201

The referenced commit might have revealed a problem in the libs.javafx modulesetup (javafx can't be found on the classpath):

![image](https://github.com/apache/netbeans/assets/2179736/f41636dc-7e5c-4aa1-bf7b-8c1b60bbdcb1)

**Comment by matthiasblaesing** [comment]
@sdedic could you please have a look at this?

**Comment by neilcsmith-net** [comment]
I do wish we'd not rushed that one in after freeze. Cause of a few headaches. Third possible fix for an 18-u1?

Aside - you can use the task / meta form and relabel as a bug to bypass the main issue form, or even just use issues/new in the address bar - save emailing everyone the asdf. 😉

**Comment by matthiasblaesing** [comment]
Yeah sorry for the noise. I was mildly annoyed as this took me about 2-3 hours bisecting. Just read this in the PR cover letter:

> For autoload modules that provide tokens, the same check as for eager modules was added: this allows early detection of autoloads with unsatisfied java/package dependencies such as graalvm.sdk and 'javafx.linux`.

I think the issue that was overlooked here was, that on JDKs 11+ two modules are potentially required to be loaded together. 

`org.netbeans.libs.javafx` is the base module, that is always required. On Oracle JDK 8 this module alone is enough as JavaFX is loaded from the `ext` directory (via the `Classpath` entry in the manifest). On JDK 11+ though one of the `org.netbeans.libs.javafx.<OS>` modules needs to be loaded. These modules are declared as fragments loaded into the host `org.netbeans.libs.javafx`. This special case (fragements) has to be considered when evaluating loading problems.

For OpenJFX we can ditch this with NB19, but NB18 should still support JDK 8.

**Comment by mbien** [comment]
this might be related to #5955, since tests started to have issues loading javafx too. It seems to be a race condition though, success chance is fairly low.

I tried to revert the complete changeset of 5842 to run the platform tests in CI and check if that stabilizes the tests but it conflicts unfortunately (and have no time for this now).

**Comment by MartinBalin** [comment]
We cannot revert the fix as it will mean JS is not loaded and proxy autodetection does not work again. JavaFX not loading is serious problem and needs to be fixed.

**Comment by sdedic** [comment]
@neilcsmith-net the possible fix should go to `release180`, right ?

**Comment by neilcsmith-net** [comment]
@sdedic to master, then once verified, cherry picked in a separate PR to release180.

**Comment by sdedic** [comment]
OK, the culprit is ModuleManager:2199. I've consulted that with @jtulach and we come to conclusion that the Package dependency was originally meant as a test for system package. Otherwise it would require, at the point of this getProblems() test, to bring up classloaders for the entire subtree rooted at the tested module: this is both barely possible (as dependencies are not deep-discovered yet, entire `simulateEnable` uses depth-first traversal, and terrible performance.

If the check is disabled at this point (as it was earlier), module system might accept a module (which will be later rejected for missing required package) - and will load *all its dependencies*. While the broken module will not load in the final stage, all its dependencies will, recursively, even though they are not needed in the system. Those dependencies may register services, `-Provides` declarations etc, further polluting the module system.

In addition, NB prints an error on startup that a module cannot be enabled bcs. of missing dependency - and not only that: the failed module actually breaks the startup, if there's > 1 such module (one fallback is attempted).

**Comment by sdedic** [comment]
@matthiasblaesing  ?
